packages/harmony/src/components/checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,125 @@
+import { ChangeEvent, forwardRef, Ref, useCallback } from 'react'
+
+import { useTheme } from 'foundations'
+import { useControlled } from 'hooks/useControlled'
+
+import { Flex, Box } from '../layout'
+
+import IconIndeterminate from './Indeterminate.svg'
+import IconSelect from './Select.svg'
+import { CheckboxProps } from './types'
+
+/*
+ * An input for choosing from predefined options:
+ * (1) when used alone, it gives a binary choice (checked/unchecked)
+ * (2) in a group it allows the user to select multiple values from a list of options.
+ */
+export const Checkbox = forwardRef(
+  (props: CheckboxProps, ref: Ref<HTMLInputElement>) => {
+    const {
+      checked: checkedProp,
+      defaultChecked,
+      indeterminate,
+      onChange,
+      _isFocused,
+      _isHovered,
+      ...other
+    } = props
+    const { color, motion } = useTheme()
+
+    const [checked, setChecked] = useControlled({
+      controlledProp: checkedProp,
+      defaultValue: defaultChecked,
+      stateName: 'checked',
+      componentName: 'Checkbox'
+    })
+
+    const handleChange = useCallback(
+      (e: ChangeEvent<HTMLInputElement>) => {
+        onChange?.(e)
+        setChecked(e.target.checked)
+      },
+      [onChange, setChecked]
+    )
+
+    const Icon = !checked
+      ? null
+      : indeterminate
+      ? IconIndeterminate
+      : IconSelect
+
+    const hoverCss = {
+      backgroundColor: checked ? color.secondary.s200 : color.neutral.n150
+    }
+
+    const focusCss = {
+      borderRadius: 6,
+      border: `2px solid ${color.border.accent}`
+    }
+
+    return (
+      <Flex
+        h='unit7'
+        w='unit7'
+        as='span'
+        alignItems='center'
+        justifyContent='center'
+      >
+        <input
+          ref={ref}
+          type='checkbox'
+          css={{
+            position: 'absolute',
+            top: 0,
+            left: 0,
+            opacity: 0,
```

Comment:
is opacity 0 not the default?

Reply:
naw, so its hidden in this case, but covers the full checkbox area so "onChange" is called!